types/index.d.ts

```diff
@@ -0,0 +1,279 @@
+// TypeScript Version: 2.4
+
+import * as React from 'react'
+
+/* Utils */
+
+export type Updater<T> = (value: T | ((updater: T) => T)) => void
```

Comment:
Can you check that this work correctly. T may be inferred like any function including function and function signature won't be considered. We fixed it in flow like this
```
export type Updater<T> = (value: ((updater: T) => T) | T) => void
```

Reply:
Not a problem for TS. `T` is infered from "caller" generic.
Anyway - I've added a few tests.